OBPIH-4710 Create reusable filter component

kchelstowski opened the pull request.

drodzewicz commented on src/js/components/Filter/FilterForm.jsx:
```diff
@@ -0,0 +1,115 @@
+
+              </div>
+              <div className="submit-buttons">
+                <button
```
Regarding the custom`Button` component that you would be able to use here

kchelstowski replied:
Oh, I forgot to delete this section as in the beginning I was just copying form from some other component. It should be (and will be) deleted of course.

drodzewicz commented on src/js/components/Filter/FilterForm.jsx:
```diff
@@ -0,0 +1,115 @@
+  const [amountFilled, setAmountFilled] = useState(0);
+  const countFilled = (values) => {
+    // Calculate which object's values are not empty
+    setAmountFilled(Object.keys(values).filter(key => values[key]).length);
```
A better solution would probably be to do `Object.values`
`Object.values(values).filter((val) => val).length`

drodzewicz replied:
but also remember about `falsy` values like `0`.
If value of a filter will be `0` it will result as `false` during the filter and will not be counted

kchelstowski replied:
that's nice point, I probably didn't consider this as I was only using string-like fields for testing. Thanks for that

kchelstowski commented on src/js/components/form-elements/ButtonPrimary.jsx:
```diff
@@ -0,0 +1,22 @@
+
+import Translate from 'utils/Translate';
+
+const ButtonPrimary = ({ label, defaultLabel, disabled }) => (
```
Yeah, I agree but wanted to split hem, but at one point thought that if they may be easier to use if they are separated.
As for second comment - they are global, I don't know why you think they aren't?

kchelstowski commented:
@drodzewicz  I made `<Button>` instead of two separate `<ButtonPrimary>` `<ButtonTransparent>` and also fixed the issues you mentioned.
I also removed `<Button>` from `FilterForm.md` and added it to separate `.md` file inside `form-elements`.
